portfolio/src/main/java/com/google/sps/validators/TextValidator.java

```diff
@@ -0,0 +1,37 @@
+package com.google.sps.validators;
+
+import com.google.sps.common.Constants;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class TextValidator {
+  // Singleton Design Pattern
+  private static TextValidator validator; 
+
+  // Allow only alphanumeric chars and few extra symbols.
+  private static final Pattern validPattern = Pattern.compile(Constants.TEXT_VALID_REGEX);
+
+  public enum INPUT_TYPE { NAME, COMMENT; }
```

Comment:
enums follow the naming principles for classes

https://google.github.io/styleguide/javaguide.html#s5-naming